Source/control.cpp

```diff
@@ -1690,24 +1671,9 @@ void ReleaseChrBtns(bool addAllStatPoints)
 			int statPointsToAdd = 1;
 			if (addAllStatPoints)
 				statPointsToAdd = CapStatPointsToAdd(myPlayer._pStatPts, myPlayer, attribute);
-			switch (attribute) {
-			case CharacterAttribute::Strength:
-				NetSendCmdParam1(true, CMD_ADDSTR, statPointsToAdd);
-				myPlayer._pStatPts -= statPointsToAdd;
-				break;
-			case CharacterAttribute::Magic:
-				NetSendCmdParam1(true, CMD_ADDMAG, statPointsToAdd);
-				myPlayer._pStatPts -= statPointsToAdd;
-				break;
-			case CharacterAttribute::Dexterity:
-				NetSendCmdParam1(true, CMD_ADDDEX, statPointsToAdd);
-				myPlayer._pStatPts -= statPointsToAdd;
-				break;
-			case CharacterAttribute::Vitality:
-				NetSendCmdParam1(true, CMD_ADDVIT, statPointsToAdd);
-				myPlayer._pStatPts -= statPointsToAdd;
-				break;
-			}
+			_cmd_id cmd[] = { CMD_ADDSTR, CMD_ADDMAG, CMD_ADDDEX, CMD_ADDVIT };
+			NetSendCmdParam1(true, cmd[static_cast<int>(attribute)], statPointsToAdd);

```

Comment:
Please avoid using the underlying value of the enum like this.

Reply:
be glad I didn't do (CMD_ADDSTR + static_cast<int>(attribute)) : P

Reply:
This is a common pattern in the Linux kernel. However, Linux kernel uses the enum-keyed arrays GCC/clang extension, making this robust.

In our case, it be better to extract the mapping into a function with a `switch`.